onnxruntime/test/framework/ep_plugin_provider_test.cc

```diff
@@ -317,4 +334,121 @@ TEST(PluginExecutionProviderTest, InferOrtDeviceFromDeviceMemoryInfo) {
+      st != nullptr) {
+    return st;
+  }
+
+  return nullptr;
+}
+
+// Tests that GetCapability() doesn't crash if a plugin EP tries to claim only nodes that are
+// already assigned to another EP.
+TEST(PluginExecutionProviderTest, GetCapability_ClaimOnlyNodesAssignedToOtherEP) {
+  auto [ep, ort_ep] = test_plugin_ep::MakeTestOrtEp();
+  ort_ep->GetCapability = GetCapabilityTakeAllNodes;
+
+  // Load a model and forcibly assign the only Mul node to another EP named 'OtherEp'.
+  std::shared_ptr<Model> model;
+  Status status = Model::Load(ORT_TSTR("testdata/mul_1.onnx"), model, nullptr, DefaultLoggingManager().DefaultLogger());
+  ASSERT_TRUE(status.IsOK());
+
+  Graph& graph = model->MainGraph();
+  for (Node& node : graph.Nodes()) {
+    node.SetExecutionProviderType("OtherEp");
+  }
+
+  std::filesystem::path log_file = ORT_TSTR("log_get_capability.txt");
+  if (std::filesystem::exists(log_file)) {
+    std::filesystem::remove(log_file);
+  }
+
+  // Call IExecutionProvider::GetCapability. The underlying OrtEp will try to take all nodes.
+  // Should not crash and should return an empty result.
+  {
+    logging::LoggingManager log_manager{std::make_unique<logging::FileSink>(log_file, false, false),
+                                        logging::Severity::kWARNING, false,
+                                        logging::LoggingManager::InstanceType::Temporal};
+    auto file_logger = log_manager.CreateLogger("FileLogger");
+    ep->SetLogger(file_logger.get());  // Make EP log to a file.
+
+    auto compute_capabilities = ep->GetCapability(GraphViewer(graph),
+                                                  test_plugin_ep::MockKernelLookup{},
+                                                  GraphOptimizerRegistry(nullptr, nullptr, file_logger.get()),
+                                                  nullptr);
+    EXPECT_TRUE(compute_capabilities.empty());  // No compute capabilities returned.
+  }
+
+  ASSERT_TRUE(std::filesystem::exists(log_file));
+  CheckStringInFile(log_file, "Found one or more nodes that were already assigned to a different EP named 'OtherEp'");
+
+  if (std::filesystem::exists(log_file)) {
+    std::filesystem::remove(log_file);
+  }
+}
+
+// Tests that GetCapability() doesn't crash if a plugin EP tries to claim a mix of unassigned nodes and
+// nodes that are already assigned to another EP.
+TEST(PluginExecutionProviderTest, GetCapability_ClaimSomeNodesAssignedToOtherEP) {
+  auto [ep, ort_ep] = test_plugin_ep::MakeTestOrtEp();
+  ort_ep->GetCapability = GetCapabilityTakeAllNodes;
+
+  // Load a model and forcibly assign only the first Add node to another EP named 'OtherEp'.
+  // Other nodes are unassigned and should be taken by the test plugin EP.
+  std::shared_ptr<Model> model;
+  Status status = Model::Load(ORT_TSTR("testdata/add_mul_add.onnx"), model, nullptr, DefaultLoggingManager().DefaultLogger());
+  ASSERT_TRUE(status.IsOK());
```

Comment:
nit: ASSERT_STATUS_OK to also output any error message.

Reply:
Updated, thanks